api/migrations/20210316214525-addIssuingAgencyAndRoleToHistoricalNRCED.js

```diff
@@ -0,0 +1,182 @@
+'use strict';
+
+const { ApplicationRoles, ApplicationAgencies } = require('../src/utils/constants/misc');
+
+const LegislationActs = {
+  ACT_Agri_Land_Commission: 'Agricultural Land Commission Act',
+  ACT_Dike_Maintenance: 'Dike Maintenance Act',
+  ACT_Env_Management: 'Environmental Management Act',
+  ACT_Fish_Seafood: 'Fish and Seafood Act',
+  ACT_Fisheries: 'Fisheries Act (Canada)',
+  ACT_Food_Safety: 'Food Safety Act',
+  ACT_Int_Pest_Management: 'Integrated Pest Management Act',
+  ACT_Mines: 'Mines Act',
+  ACT_Oil_Gas_Activities: 'Oil and Gas Activities Act',
+  ACT_Park: 'Park Act',
+  ACT_Water: 'Water Act',
+  ACT_Water_Sustainability: 'Water Sustainability Act',
+  ACT_Wildlife: 'Wildlife Act'
+};
+
+var dbm;
+var type;
+var seed;
+let recordCount = 0;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function(options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = async function(db) {
+  console.log('**** Adding Issuing Agency to historical NRCED records ****');
+  const mClient = await db.connection.connect(db.connectionString, { native_parser: true });
+
+  try {
+    console.log('attempting migration...');
+
+    const nrpti = await mClient.collection('nrpti');
+
+    const ocersRecords = await nrpti.find({ sourceSystemRef: 'ocers-csv' }).toArray();
+
+    await ocersRecords.forEach(async record => {
```

Comment:
The `await` here isn't actually waiting for the DB updates to finish.  One way around this is to convert the `forEach` to a `for...of` loop and await on the DB update code.

The `db.update` calls in `updateRecord` should have await, and `updateRecord` should be async.  Then each of the `updateRecord` calls should also have await.

I'd also use `updateOne` or `updateMany`.  `update` seems to be a deprecated function.